packages/catalog-model/src/location/validation.ts

```diff
@@ -34,3 +34,10 @@ export const locationSchema = yup
   })
   .noUnknown()
   .required();
+
+export const repoPathSchema = yup
```

Comment:
I believe the input would be 

```ts
type AnalyzeLocationRequest = {
  location: LocationSpec;
};
```

for now; there may eventually be more fields added to that but at this point a location probably suffices. There is probably no reason to invent a separate way to represent the location to be analyzed.

The output could be something like

(This is all just suggestions as a basis for discussion!)

```ts
type AnalyzeLocationResponse = {
  existingEntityFiles: AnalyzeLocationExistingEntity[];
  generateEntities: AnalyzeLocationGenerateEntity[];
}

// If the folder pointed to already contained catalog info yaml files, they are
// read and emitted like this so that the frontend can inform the user that it
// located them and can make sure to register them as well if they weren't
// already
type AnalyzeLocationExistingEntity = {
  location: LocationSpec;
  isRegistered: boolean;
  entity: Entity;
};

// This is some form of representation of what the analyzer could deduce.
// We should probably have a chat about how this can best be conveyed to
// the frontend. It'll probably contain a (possibly incomplete) entity, plus
// enough info for the frontend to know what form data to show to the user
// for overriding/completing the info.
type AnalyzeLocationGenerateEntity = {
  // Some form of partial representation of the entity
  entity: RecursivePartial<Entity>;
  // Lists the suggestions that the user may want to override
  fields: AnalyzeLocationEntityField[];
};

// This is where I get really vague. Something like this perhaps? Or it could be
// something like a json-schema that contains enough info for the frontend to
// be able to present a form and explanations
type AnalyzeLocationEntityField = {
  // e.g. "spec.owner"? The frontend needs to know how to "inject" the field into the
  // entity again if the user wants to change it
  field: string;

  // The outcome of the analysis for this particular field
  state: 'analysis_suggested_value' | 'analysis_suggested_no_value' | 'needs_user_input';

  // If the analysis did suggest a value, this is where it would be. Not sure if we want
  // to limit this to strings or if we want it to be any JsonValue
  value: string | null;

  // A text to show to the user to inform about the choices made. Like, it could say
  // "Found a CODEOWNERS file that covers this target, so we suggest leaving this
  // field empty; which would currently make it owned by X" where X is taken from the
  // codeowners file.
  description: string; 
};
```

Reply:
Added types that you suggested